packages/@lwc/ssr-runtime/src/lightning-element.ts

```diff
@@ -91,35 +76,31 @@ export class LightningElement implements PropsAvailableAtConstruction {
         return (this.#classList = new ClassList(this));
     }
 
-    #setAttribute(attrName: string, attrValue: string | null): void {
-        this.#attrs[attrName] = attrValue;
-        mutationTracker.add(this, attrName);
-    }
-
-    setAttribute(attrName: string, attrValue: unknown): void {
-        this.#setAttribute(attrName, String(attrValue));
+    setAttribute(attrName: unknown, attrValue: unknown): void {
+        if (typeof attrName === 'string') {
```

Comment:
Good catch. I looked into [XML name validation](https://github.com/jsdom/xml-name-validator) but I think it might be overkill. If we want it, we should add it sooner than later before component authors take a dependency on the proposed behavior (or is it too late already?).